cmds/ping/ping.go

```diff
@@ -69,47 +76,96 @@ func beatifulLatency(before time.Time) (latency string) {
 	return latency
 }
 
+func cksum(bs []byte) uint16 {
+	sum := uint32(0)
+
+	for k := 0; k < len(bs)/2; k++ {
+		sum += uint32(bs[k*2]) << 8
+		sum += uint32(bs[k*2+1])
+	}
+	if len(bs)%2 != 0 {
+		sum += uint32(bs[len(bs)-1]) << 8
+	}
+	sum = (sum >> 16) + (sum & 0xffff)
+	sum = (sum >> 16) + (sum & 0xffff)
+	if sum == 0xffff {
+		sum = 0
+	}
+
+	return ^uint16(sum)
+}
+
+func ping1(netname string, host string, i uint64) {
```

Comment:
in general a function like this should return an error and let the parent make the decision.

Reply:
Ok. Done.